ocaml/database/db_cache_types.ml

```diff
@@ -351,6 +351,7 @@ let remove_from_set key t =
 
 exception Duplicate
 let add_to_map ~idempotent key value t =
+  if String.length key = 0 then raise Db_exn.Empty_key_in_map;
```

Comment:
This is more expensive than necessary because it always takes the string length. It's cheaper to test `if key = "" then .. `

Reply:
https://github.com/ocaml/ocaml/blob/4.04/byterun/str.c#L39-L45

Reply:
I'm not sure which is cheaper to be honest.

Reply:
It's fine as it is because this operation is cheap, as your link to the C is showing.